Assignment 2.Rmd

```diff
@@ -80,15 +91,76 @@ plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender)
 
 plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.width=EDGE$count)
-````
+```
 
 ## Part II
 
 In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
 
+```{r}
+library(dplyr)
+t_count <- aggregate(EDGE$count, by=list(from=EDGE$from), FUN=sum)
+names(t_count) <- c("id", "count")
+st_count <- aggregate(EDGE$count, by=list(from=EDGE$to), FUN=sum)
+names(st_count) <- c("id", "count")
+VERTEX <- left_join(VERTEX, rbind(t_count, st_count), by="id")
+plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.width=EDGE$count, edge.width=EDGE$count, vertex.size=VERTEX$count)
+```
+
+
 ## Part III
 
 Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
 
-Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweeness centrality and dregree. **Who is the most central person in the network?**
+```{r}
+#Libraries
+library(tidyr)
+library(dplyr)
+library(igraph)
+
+#Data Maniputation
+D3 <- read.csv("~/Desktop/HUDK4050/assignment2/HUDK4050-2018-classes.csv", header = FALSE)
+D3 <- D3[-c(1,3), ]
+names(D3) <- as.matrix(D3[1, ])
+D3 <- unite(D3, Name, c(`First Name`, `Last Name`), remove=TRUE)
+D3 <- data.frame(D3[-1, 15:20])
+
+#Combine the course columns into one
+D4 <- gather(D3, Class, Course, `Class.1`:`Class.5`, na.rm = TRUE) 
+D4$Course <- gsub(" ", "", D4$Course)
+D4 <- select(D4, Name, Course) %>% filter(Course > 0, Course!="HUDK4050")
+
+#Add the count variable
+D4$Count <- 1
+
+#Convert to the person-course matrix
+D5 <- spread(D4, Course, Count) %>% as.matrix()
+
+#Create the person-person matrix
+row.names(D5) <- D5[, 1]
+D5 <- D5[, -1]
+#D5[is.na(D5)]<- 0
+D5 <- ifelse(is.na(D5), 0, 1)
+D6 <- D5%*%t(D5)
+diag(D6) <- 0
+
+#Visualization
+graph <- graph.adjacency(D6, mode="undirected")
+plot(graph, layout=layout.fruchterman.reingold)
+
+```
+
+
+Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweeness centrality and degree. **Who is the most central person in the network?**
+```{r}
+#Betweeness centrality
+betweenness(graph, v = V(graph), directed = FALSE, weights = NULL,
+  nobigint = TRUE, normalized = FALSE)
+
+#Degree
+degree(graph, v = V(graph), mode = c("all", "out", "in", "total"),
+  loops = TRUE, normalized = FALSE)
+
+#Conclusion: Xixuan Zhao is the most central person in network.
```

Comment:
This is great. In the future, can you think of a way to automatically pull out the person who is the most central? Also, what does it mean to be the most central?